notebooks/metrics_demo.ipynb

```diff
@@ -88,7 +93,10 @@
     "## Trustworthiness\n",
     "It is a measure of the extent to which the local structure is retained in the embedding of the model. Therefore, if a sample predicted by the model lied within the unexpected region of the nearest neighbors, then those samples would be penalized. For more information on the trustworthiness metric please refer to: https://scikit-learn.org/dev/modules/generated/sklearn.manifold.t_sne.trustworthiness.html\n",
     "\n",
-    "the documentation for cuML's implementation of the trustworthiness metric is: https://rapidsai.github.io/projects/cuml/en/stable/api.html#cuml.metrics.trustworthiness.trustworthiness\n"
+    "the documentation for cuML's implementation of the trustworthiness metric is: https://rapidsai.github.io/projects/cuml/en/stable/api.html#cuml.metrics.trustworthiness.trustworthiness\n",
```

Comment:
Since we are combining multiple demos into a single end-to-end notebook, I think it would be easier for users to follow if we separated the data generation, model building and model evaluation for each algorithm.

An example flow could be:

## Linear Regression 
1. Briefly describe the demo, including datasets, estimator, and evaluation criteria.
2. Build train and test datasets (use generator and train/test split, etc...)
3. Train model
4. Evaluate model, perhaps using a cross-validation on the R2 score (the cross-val would be an opportunity to show how our estimators are scikit-learn compatible).

## Clustering
1. Briefly describe the demo, including datasets, estimator, and evaluation criteria
2. Build train and test datasets
3. Train model
4. Evaluate model

## Classification
1. Briefly describe demo
.....
4. Evaluate model. Perhaps demonstrate our new confusion matrix / AUC

This type of flow will allow users to jump right to the part of interest and get a full demo, without having to jump around the notebook to look up variable names. It will also allow them to play with the different data and hyperparameters right in one place, rather than having to jump around.

Reply:
If we end up structuring the notebook in this way, we probably want to describe this layout at the very beginning. As @miguelangel has pointed out in the past, if we end up breaking these algorithms up into separate sections, we might also want to consider keeping the imports close to where they are used to make it easier for users to copy/paste.